### ISSUE TITLE
Cannot create ClickHouse dataset when database name contains dash

### ISSUE THREAD

**Issue by uranuswch** [apache/superset#24372]
I use docker-compose to deploy locally, want to connect ClickHouse with a database named `dev-db`.

But its can not scan the schema, I think it because there has a dash `-` in the database name.

Console error log: 
```
ERROR:clickhouse_connect.driver.httpclient:Code: 62. DB::Exception: Syntax error: failed at position 21 ('-') (line 1, col 21): -db
FORMAT Native. Expected one of: NOT, ILIKE, LIKE, WHERE, LIMIT, INTO OUTFILE, FORMAT, SETTINGS, end of query. (SYNTAX_ERROR) (version 23.3.2.37 (official build))
```

#### How to reproduce the bug

1. Go to 'Create Dataset' and select ClickHouse Database contains a dash.

#### Screenshots

<img width="303" alt="CleanShot 2023-06-13 at 13 23 47@2x" src="https://github.com/apache/superset/assets/9143888/b9e7b4a6-8dbf-4320-acba-fffe8ef5577d">

<img width="390" alt="CleanShot 2023-06-13 at 13 36 21@2x" src="https://github.com/apache/superset/assets/9143888/103a1987-7d2f-40e9-9e9a-719e1f3579f7">

### Environment

- browser type and version: Safari 16.5 and Chrome 114.0.5735.106
- superset version: `lastest-dev`
- python version: 3.11.3
- node.js version: 20.2.0
- any feature flags active: clickhouse-connect v0.6.2

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by kovitals** [comment]
Can't reproduce on version 2.0.0, works as expected

`create database if not exists "dev-db";

create table "dev-db".test (
    col1 String,
    col2 Int16
) engine = Log();`

<img width="866" alt="image" src="https://github.com/apache/superset/assets/630890/0c75ae2b-a083-44cd-a4a4-e2f100fc8d88">

**Comment by uranuswch** [comment]
Thanks for your reply.

I switch to the version 2.0.0, it works correctly on fetching tables, but it seems not support all the ClickHouse types, the console reported errors like `Failed to execute query '3' - 'None': 'NoneType' object has no attribute 'strip'` 

:(

So at last I have to switch to the latest version and create new database with alias view to walkaround it.

Btw, I checked the two versions: 2.1.0(latest dev) vs 2.0.0, the difference between them:
- 2.0.0 use `clickhouse-driver` and `clickhouse-sqlalchemy` to connect CH.
- 2.1.0 use `clickhouse-connect` to connect CH.

It seems have a bug on `clickhouse-connect` when fetching table which name has a dash, hopes you can fix it in the future.

Many thanks.

**Comment by genzgd** [comment]
You can use either clickhouse-driver/clickhouse-sqlalchemy or clickhouse-connect with Superset.  If you install those connectors, both should be available as data sources.

I'll take a look at the issue with fetching tables using clickhouse-connect.

**Comment by uranuswch** [comment]
> You can use either clickhouse-driver/clickhouse-sqlalchemy or clickhouse-connect with Superset. If you install those connectors, both should be available as data sources.
> 
> I'll take a look at the issue with fetching tables using clickhouse-connect.

Thanks, I will try driver/sqlalchemy later.

**Comment by genzgd** [comment]
I couldn't reproduce an issue on recent dev version of Superset.  Can you provide the `CREATE TABLE` statement and query that fails?

**Comment by uranuswch** [comment]
> I couldn't reproduce an issue on recent dev version of Superset. Can you provide the `CREATE TABLE` statement and query that fails?

```
CREATE DATABASE IF NOT EXISTS `dev-db` ON CLUSTER default;

CREATE TABLE IF NOT EXISTS `log_events` ON CLUSTER default (
    `_timestamp` DateTime64(3, 'UTC') COMMENT 'timestamp of event',
    `number_names` Array(String) COMMENT 'the name of numeric field in attributes',
    `number_values` Array(DECIMAL256(6)) COMMENT 'the value of numeric field in attributes',
    `string_names` Array(String) COMMENT 'the name of string field in attributes',
    `string_values` Array(String) COMMENT 'the value of string field in attributes',
    `boolean_names` Array(String) COMMENT 'the name of boolean field in attributes',
    `boolean_values` Array(UInt8) COMMENT 'the value of boolean field in attributes',
    `time_names` Array(String) COMMENT 'the name of time field in attributes',
    `time_values` Array(DateTime64(3, 'UTC')) COMMENT 'the value of time field in attributes',
    `array_names` Array(String) COMMENT 'the name of array field in attributes',
    `array_values` Array(Array(String)) COMMENT 'the value of array field in attributes',
) ENGINE = ReplicatedMergeTree('/clickhouse/tables/{database}/{shard}/log_events', '{replica}')
PARTITION BY toYYYYMM(`_timestamp`)
ORDER BY (toYYYYMMDD(_timestamp))
SETTINGS index_granularity = 8192;
```

You can follow this video to reproduce.

https://github.com/apache/superset/assets/9143888/27e40b43-4fd2-4d5a-a176-c1380ce8d133

**Comment by genzgd** [comment]
Thanks!  I've reproduced and there will be a fix in the next patch release (0.6.4) of clickhouse-connect.  It will probably be out next week some time.
